File: 0435-non-overlapping-intervals/solution.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        if not intervals: return 0
        intervals.sort(key=lambda i:i[0])
        res = 0
        prevRight = intervals[0][1]
        for i in range(1, len(intervals)):
            if intervals[i][0] < prevRight:
                res += 1
                prevRight = min(prevRight, intervals[i][1])
            else:
                prevRight = intervals[i][1]
        return res

File: 0435-non-overlapping-intervals/solution.cpp
class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        if(intervals.size() <= 1){
            return 0;
        }
        sort(intervals.begin(), intervals.end());
        int res = 0;
        auto prev = intervals[0];
        for(int i = 1; i < intervals.size(); ++i){
            auto cur = intervals[i];
            if(prev[1] > cur[0]){
                res++;
                prev[1] = min(cur[1], prev[1]);
            }
            else{
                prev = cur;
            }

        }
        return res;
    }
};
